examples/typescript/swap.ts

```diff
@@ -0,0 +1,275 @@
+/**
+ * Example to demonstrate creating and adding to liquidity pools, swapping between two fungible asset.
+ *
+ * Steps:
+ * 1. Create two accounts (Alice and Bob)
+ * 2. Fund Alice and Bob
+ * 3. Create a FA with Alice as the owner
+ * 4. Create another FA with Bob as the owner
+ * 5. Create a liquidity pool with Alice's FA and Bob's FA
+ * 6. Swap between Alice's FA and Bob's FA
+ */
+
+import {
+  Account,
+  AccountAddress,
+  Aptos,
+  Bool,
+  Ed25519PrivateKey,
+  Hex,
+  HexInput,
+  MoveObject,
+  MoveString,
+  U64,
+  ViewRequestData,
+} from "aptos";
+
+const readline = require("readline").createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const getOptimalLpAmount = async (
+  aptos: Aptos,
+  swap: AccountAddress,
+  token1Addr: HexInput,
+  token2Addr: HexInput,
```

Comment:
We should probably take a look into making a proper AccountAddress input (AccountAddress | HexString), since I suspect people won't be passing in byte arrays for the account address.  Since the swap is an `AccountAddress` having this mixture of types may be a little strange.

Reply:
+1. If we can minimize the need to know how all these similar types differ, it'd be great!